src/util.cc

```diff
@@ -423,21 +427,79 @@ const char* SpellcheckString(const char* text, ...) {
 }
 
 #ifdef _WIN32
+wstring Utf8ToWide(const string& u8_string) {
+  if (u8_string.empty())
+    return wstring();
+
+  size_t size_needed = MultiByteToWideChar(CP_UTF8, 0, u8_string.data(),
```

Comment:
Instead of using an std::string, use Ninja's internal string_view look-alike, `StringPiece`.